src/systems/ec/implementations/vanilla/packages/ec-component-gallery/ec-component-gallery.js

```diff
@@ -205,18 +209,91 @@ export class Gallery {
 
   updateOverlay(selectedItem) {
     this.selectedItem = selectedItem;
+    const video = queryOne('video', selectedItem);
+    let mediaElement = null;
+
+    // Update media
+    if (video != null) {
+      // Media is a video
+      mediaElement = document.createElement('video');
+      mediaElement.setAttribute('poster', video.poster);
+      mediaElement.setAttribute('controls', 'controls');
+      mediaElement.classList.add('ecl-gallery__slider-video');
+
+      if (this.overlayMedia) {
+        this.overlayMedia.innerHTML = '';
+        /* eslint-disable-next-line unicorn/prefer-node-append */
+        this.overlayMedia.appendChild(mediaElement);
+      }
+
+      // Get sources
+      const sources = queryAll('source', video);
+      sources.map(source => {
```

Comment:
it looks like what you need is a `forEach`, not a `map`

Reply:
as far as I know, forEach are not supported by IE